lib/nerves_motd.ex

```diff
@@ -178,17 +178,62 @@ defmodule NervesMOTD do
     Enum.reject([days, hours, minutes, seconds], &is_nil/1)
   end
 
-  @spec clock() :: [binary(), ...]
+  @spec clock() :: binary() | [binary()]
   defp clock() do
+    local_time(NervesTimeZones) || local_time(NaiveDateTime) || utc_time()
+  end
+
+  # 2021-10-30 11:50:09-04:00 EDT America/New_York
+  defp local_time(NervesTimeZones) do
+    case Code.ensure_loaded(NervesTimeZones) do
+      {:module, mod} ->
+        mod.get_time_zone()
+        |> DateTime.now!()
+        |> DateTime.truncate(:second)
+        |> DateTime.to_string()
+
+      _ ->
+        nil
+    end
+  end
+
+  # 2021-10-30 11:49:44-0400 EDT
+  defp local_time(NaiveDateTime) do
+    if Version.match?(System.version(), "~> 1.10") do
+      case Code.ensure_loaded(NaiveDateTime) do
+        {:module, mod} ->
+          [
+            mod.local_now()
+            |> NaiveDateTime.truncate(:second)
+            |> NaiveDateTime.to_string(),
+            system_time_zone_name()
+          ]
+
+        _ ->
+          nil
+      end
+    end
+  end
+
+  # 2021-10-30 15:49:30+0000 UTC
+  defp utc_time() do
     [
       DateTime.utc_now()
       |> DateTime.truncate(:second)
       |> DateTime.to_string()
       |> String.trim_trailing("Z"),
-      " UTC"
+      "+0000 UTC"
     ]
   end
 
+  defp system_time_zone_name() do
+    # -0400 EDT
+    case System.cmd("date", ["+%z %Z"]) do
```

Comment:
That's clever. 

I looked at the Erlang/OTP source code to see if there's an equivalent to this, but there's not. :(

Practically speaking, though, this will only return UTC on Nerves. The reason is that it requires the `TZ` environment variable to be set before Erlang starts. Someone could do that with `erlinit.conf` and supply tzdata files, but that seems like a lot of work for hardcoding a timezone. `NervesTimeZone` would be way easier.